### PR TITLE
RPP BitwiseAND and BitwiseOR Tensor on HOST and HIP

### PR DESCRIPTION
**Pull request by r-abishek**
- Adds support for U8/F16/F32/I8 BitwiseAND/BitwiseOR on HOST and HIP
- Adds relevant unit/perf/QA tests

### CODE REVIEW

**Review by randyh62 — APPROVED** [review]
Reviewed text changes, related to License updates. Seems fine. Though, there is a lot of content here. Not sure if I saw it all.

File: src/include/hip/rpp_hip_common.hpp [hunk 1]
```diff
@@ -1713,6 +1713,34 @@ __device__ __forceinline__ void rpp_hip_math_multiply24_const(d_float24 *src_f24
     dst_f24->f4[5] = src_f24->f4[5] * multiplier_f4;
 }
 
+// d_float8 bitwiseAND
+
+__device__ __forceinline__ void rpp_hip_math_bitwiseAnd8(d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8)
+{
+        dst_f8->f1[0] = (float)((uchar)(src1_f8->f1[0]) & (uchar)(src2_f8->f1[0]));
```
Comment (rrawther) [inline src/include/hip/rpp_hip_common.hpp:1720]: why  can't we do it as vector instructions?

File: src/modules/cpu/kernel/bitwise_and.hpp [hunk 2]
```diff
@@ -0,0 +1,965 @@
...
+                    srcPtr1Temp += vectorIncrement;
+                    srcPtr2Temp += vectorIncrement;
+                    dstPtrTempR += vectorIncrementPerChannel;
+                    dstPtrTempG += vectorIncrementPerChannel;
+                    dstPtrTempB += vectorIncrementPerChannel;
+                }
+
+                for (; vectorLoopCount < bufferLength; vectorLoopCount += 3)
+                {
+                    *dstPtrTempR++ = static_cast<Rpp8u>(RPPPIXELCHECK((srcPtr1Temp[0] & srcPtr2Temp[0])));
```
Comment (rrawther) [inline src/modules/cpu/kernel/bitwise_and.hpp:106]: I think RPPPIXELCHECK and static_cast is un-necessary here

File: src/modules/cpu/kernel/bitwise_and.hpp [hunk 3]
```diff
@@ -0,0 +1,965 @@
...
+                    srcPtr1TempB += vectorIncrementPerChannel;
+                    srcPtr2TempR += vectorIncrementPerChannel;
+                    srcPtr2TempG += vectorIncrementPerChannel;
+                    srcPtr2TempB += vectorIncrementPerChannel;
+                    dstPtrTemp += vectorIncrement;
+                }
+
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    dstPtrTemp[0] = static_cast<Rpp8u>(RPPPIXELCHECK((*srcPtr1TempR & *srcPtr2TempR)));
```
Comment (rrawther) [inline src/modules/cpu/kernel/bitwise_and.hpp:168]: same comment as above

File: src/modules/cpu/kernel/bitwise_and.hpp [hunk 4]
```diff
@@ -0,0 +1,965 @@
...
+                    srcPtr1Temp += vectorIncrement;
+                    srcPtr2Temp += vectorIncrement;
+                    dstPtrTempR += vectorIncrementPerChannel;
+                    dstPtrTempG += vectorIncrementPerChannel;
+                    dstPtrTempB += vectorIncrementPerChannel;
+                }
+#endif
+                for (; vectorLoopCount < bufferLength; vectorLoopCount += 3)
+                {
+                    *dstPtrTempR++ = RPPPIXELCHECKF32((float)((uint)(srcPtr1Temp[0] * 255) & (uint)(srcPtr2Temp[0] * 255)) / 255);
```
Comment (rrawther) [inline src/modules/cpu/kernel/bitwise_and.hpp:335]: what is the multiplication and division by 255 for?

Comment (snehaa8) [reply src/modules/cpu/kernel/bitwise_and.hpp:335]: Bitwise Operations work only on integer type inputs, thus converted float into integer then reverted it back to float.

File: src/include/hip/rpp_hip_common.hpp (hunk 1 above)
Comment (snehaa8) [reply src/include/hip/rpp_hip_common.hpp:1720]: HIP CUDA doesn't support vectorized 8 pixel instruction for Bitwise operations. Thus used pixel-wise operations.

File: src/modules/cpu/kernel/bitwise_and.hpp (hunk 4 above)
Comment (snehaa8) [reply src/modules/cpu/kernel/bitwise_and.hpp:335]: Link: [https://stackoverflow.com/questions/1723575/how-to-perform-a-bitwise-operation-on-floating-point-numbers]

File: src/modules/cpu/kernel/bitwise_and.hpp (hunk 2 above)
Comment (snehaa8) [reply src/modules/cpu/kernel/bitwise_and.hpp:106]: Modified this and created an internal PR, will be pushed in soon.

File: src/modules/cpu/kernel/bitwise_and.hpp (hunk 3 above)
Comment (snehaa8) [reply src/modules/cpu/kernel/bitwise_and.hpp:168]: Modified this too

**Pull request merged**